### DISCUSSION TITLE
Can't build iOS in Azure DevOps pipeline with anything after FB 10.15

### DISCUSSION THREAD

**Discussion by SeanVelSol** [firebase/firebase-ios-sdk#14296] [Q&A] [answered]
Can't build iOS in Azure DevOps pipeline with anything after FB 10.15 - Just as it says. Took over a project that had been using FB via SPM. We have source and CI/DC via Azure DevOps delivering to AppStore and that has been fine. However, in trying to update the packages, we've found that any version of Firebase after 10.15 will cause the pipeline to fail with a code signing error. Rolling back and it's fine. Looked at with our DevOps person, but we have yet to find a solution, so looking for any ideas or solution from the community. Doesn't seem like we would be the first to run into this but web search hasn't helped so far either. Thanks~!

**Comment by paulb777** [comment]
What is the error?

**Reply by SeanVelSol** [reply]
The XC build task is vague, any issue with signing just returns _'Error: /usr/bin/xcodebuild failed with return code: 65.'_ Removing FB or rolling the version back to 10.15 and it builds, signs and is accepted at the AppStore just fine. Looking at the invocation of xcodebuild looks correct, with the correct team/profile/etc, and as stated works without FB just fine. FWIW, we do not have problems with other swift packages in projects that do not include FB. -- EDIT: Should explicitly say, the apps build locally in Xcode, regardless of FB version, OK. The issue is only when running in DevOps pipeline.

**Comment by joran-allego** [comment]
Encountering the same problem, did you ever found a solution? I did find a bug report as well, but obviously disabling code signing is not a permanent solution and the 'name=iPhone 16' option mentioned in the last comment did not work for me either.

https://github.com/invertase/react-native-firebase/issues/7647

**Comment by joran-allego** [comment]
Update for those finding this later: for us, the issue was that Firebase added code signing capabilities to the pods, which they couldn't use. So I solved it with disabling code signings for pods again:
```
installer.pods_project.targets.each do |target|
  target.build_configurations.each do |config|
    config.build_settings['EXPANDED_CODE_SIGN_IDENTITY'] = ""
    config.build_settings['CODE_SIGNING_REQUIRED'] = "NO"
    config.build_settings['CODE_SIGNING_ALLOWED'] = "NO"
  end
end
```

**Comment by SeanVelSol** [comment, marked as answer]
The issue we ran into with the pipelines building with the latest versions of Crashlytics was because of the configuration, it was trying to sign the Google libraries as if they were our own and causing a conflict. This started showing up on Crashlytics Swift Packages after version 10.15, from what I am assuming from the errors was a switch on the library’s Xcode project from code signing NONE -> Automatic.
 
The second part, why wasn’t it happening in Xcode vs. Pipeline, is taking into consideration that the pipeline uses a Microsoft task called Xcode@5. This task calls the xcodebuild command line for you, based on the parameters you give it. And based on what you have or don’t have there it does what MS engineers thought was best – which isn’t always as expected. Trying to figure that out took some trial and error even when knowing the correct build command line format from building locally on the Mac.
 
The main issue is the way we previously passed the signing cert and profile which was basically entering at step 1 and was causing the build to think everything needed signed. That had to be removed. Now is a good time to say there are 3 steps to an iPhone app build:
 
1. Build
2. Archive - build products into a single container
3. Export - create an IPA and code sign as needed
 
After removing the code sign at the start of the process, we needed to create some plist (xml) files that are supplied to the export step 3 and ensure the Xcode@5 task was passed and configured for these instead. Once that was figured out, builds were working again. Examples below.
 
-Sean
 
 
3 Steps Command Line Examples:
 
// BUILD
/usr/bin/xcodebuild -sdk iphoneos -configuration Debug -project iosApp/iosApp.xcodeproj -scheme APP-SCHEME-NAME_QA build -verbose -destination generic/platform=iOS
 
// ARCHIVE
/usr/bin/xcodebuild -project iosApp/iosApp.xcodeproj -scheme APP-SCHEME-NAME_QA archive -sdk iphoneos -configuration Debug -archivePath /Users/username_here/Output/xctemp/test -verbose
 
// EXPORT
/usr/bin/xcodebuild -exportArchive -archivePath /Users/username_here/Output/xctemp/test.xcarchive -exportPath /Users/username_here/Output/xctemp/ -exportOptionsPlist /Users/username_here/Output/xctemp/xctempexportOptions.plist
 
 
Passed Export Options Plist Example:
 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" http://www.apple.com/DTDs/PropertyList-1.0.dtd>
<plist version="1.0">
<dict>
    <key>ApplicationProperties</key>
    <dict>
        <key>ApplicationPath</key>
        <string>Applications/APP-SCHEME-NAME_QA.app</string>
        <key>Architectures</key>
        <array>
            <string>arm64</string>
        </array>
        <key>CFBundleIdentifier</key>
        <string>com.YOURDOMAIN.APP-SCHEME-NAME.qa</string>
        <key>SigningIdentity</key>
        <string>iPhone Distribution: YOUR COMPANY NAME (YOUR-TEAM-KEY1)</string>
        <key>Team</key>
        <string>YOUR-TEAM-KEY</string>
    </dict>
    <key>provisioningProfiles</key>
    <dict>
        <key>com.YOURDOMAIN.APP-SCHEME-NAME.qa</key>
        <string>PROVISIONING-PROFILE-UUID-GOES-HERE</string>
    </dict>
    <key>Name</key>
    <string>APP-SCHEME-NAME_QA</string>
    <key>SchemeName</key>
    <string>APP-SCHEME-NAME_QA</string>
    <key>method</key>
    <string>enterprise</string>
</dict>
</plist>

**Comment by gokulkrish90** [comment]
Before joining any course, it’s better to understand the full DevOps ecosystem. This explains the complete lifecycle clearly.

👇 Sharing here
**[DevOps Training in Electronic City Bangalore](https://medium.com/@emexobangalore/best-devops-training-in-electronic-city-bangalore-certification-tools-placement-guide-2026-a0326269cd22)**
For More Details: **[DevOps Training Institute in Electronic City](https://www.emexotechnologies.com/courses/devops-training-in-electronic-city-bangalore/)**
